libraries/chain/database.cpp

```diff
@@ -143,7 +143,8 @@ void database::open( const fc::path& data_dir, const fc::path& shared_mem_dir, u
    FC_CAPTURE_LOG_AND_RETHROW( (data_dir)(shared_mem_dir)(shared_file_size) )
 }
 
-void database::reindex( const fc::path& data_dir, const fc::path& shared_mem_dir, uint64_t shared_file_size )
+void database::reindex( const fc::path& data_dir, const fc::path& shared_mem_dir, uint32_t stop_replay_at, TBenchmark benchmark,
+                        uint64_t shared_file_size, uint32_t* last_block_number /*= nullptr*/ )
```

Comment:
Let's make `last_block_number` the return type of `reindex`. We try to avoid passing pointers wherever possible.

It is a minor pet peeve, but let's put the new args at the end and default initialize them so that the previous method signature is still valid.

Reply:
Committed and pushed all requested changes. Ready to merge. #1590